### DISCUSSION TITLE
Malpositioned item in the left margin. Use of details-summary functionality.

### DISCUSSION THREAD

**Discussion by howardm** [quarto-dev/quarto-cli#3813] [Q&A]
Hi,

I'm trying to use the following functionality in a particular portion of a web page:

```
<details><summary>Content to be revealed</summary>

Content that is revealed.

</details>

```

It works, but this screenshot shows the placement of it (truncated) in the left margin

---
![Screen Shot 2022-12-31 at 7 54 02 PM](https://user-images.githubusercontent.com/6101226/210159917-0e071f62-b0e1-405c-b3fd-71c2517bbb23.jpg)

---

Is the fact that the code precedes content that includes a  `::: {.panel-tabset .column-page}` section the cause ?

This problem with the code doesn't occur elsewhere in the document.

Howard

**Comment by mcanouil** [comment]
Hi,

Could you provide a fully reproducible code example?
Without, it's really hard to reproduce, thus to fix the bug if there is indeed one.

**Comment by howardm** [comment]
I want to put the `details - summary` code immediately before this section of code:


````


### Clinical cases

<br>

A tabset for presentation of a case

This tabset presentation uses a div with the `{.column-page`} class layout mode.

<br>

[After viewing a case, return to the *Clinical Presentation* tab before starting the next one.]{.aside}


#### Case one {#sec-case-1}

<br>

::: {.panel-tabset .column-page}
## Clinical presentation

Pertinent clinical information is provided here.

## Radiography

![](images/CXR.jpg){.lightbox width="40%" fig-align="center"}

## CT

![](images/CT1.jpg){width="35%" fig-align="center"}

## CT Pacsbin

```{=html}
<div style="text-align:center;">

<iframe 
src="https://www.pacsbin.com/c/-ylmpOm7kd?s=6211650ea311df5697970ae9&i=6211650ea311df5697970afa&ww=1665&wc=-490&an=false&header=false&notes=false&overlay=false&scale=0.9" 
style="width: 475px; height: 475px; border: 0;"></iframe>

</div>
 
```
## Explanation

::: columns
::: {.column width="32%"}
![](images/CXR.jpg){.lightbox width="82%"}
:::

::: {.column width="32%"}
![](images/CT1.jpg){width="100%"}
:::

::: {.column width="2%"}
:::

::: {.column width="32%"}
Findings:

-   finding one [&xrarr;]{style="color: red"}
-   finding two
-   finding three
:::

Discussion of the case

::: {.callout-note icon="false"}
## Key Points

-   key point one
-   key point two
:::
:::

## Another Explanation example

::: columns
::: {.column width="65%"}
![](images/CXR.jpg){.lightbox width="62%"}
:::

::: {.column width="34%"}
Findings:

-   finding one  [&xrarr;]{style="color: yellow"}
-   finding two  [&xrarr;]{style="color: blue"}
-   finding three [&xrarr;]{style="color: red"}

Discussion of the case

::: {.callout-note appearance="minimal"}
Remember: Don't forgot this!
:::
:::
:::
:::

`````


(Substitute any image for : CXR and CT1 . I use the lightbox function in places.)

Let me know if more information is needed.

Thanks!

**Reply by cscheid** [reply]
We need is a full single `.qmd` file that we can render on our side so we can investigate the output.

**Comment by howardm** [comment]
Well, I can't duplicate the error again.

But I cannot get the `details` element to work for a section at all in the index.qmd file.

Here it is:  https://www.dropbox.com/sh/6mdqnw4yg7a523g/AABEoByxcFngURBFwWalPi69a?dl=0

I apologize for the confusion. I probably misunderstand the kind of content may be included in a `details` section.

Thanks.

**Reply by cscheid** [reply]
(If you are able to share your entire project, then we can attempt to reproduce on our side as well.)

**Comment by howardm** [comment]
O.K.

Here it is:  https://www.dropbox.com/sh/lkgmubyo87mc23g/AAC9pFd8afoLjTl9o1zHItx9a?dl=0

Basically, I want to "collapse" the Cases section so that each web page is not very long. I plan on showing *many* cases per page.

Maybe that's not the best idea  :-)

Howard

**Reply by cscheid** [reply]
Ok, thanks for the link to the project, I downloaded it.

I'll have to get back to you in a few days. I think your project triggered a different bug on the latest version of quarto and I'm tracking that down now.

**Reply by howardm** [reply]
Hi,

Did anyone perhaps  have the opportunity to explore this issue further ?

Cheers.

**Reply by cscheid** [reply]
Thanks for following up. I don't think I can reproduce your issue locally.

On the latest prerelease, I did have to make some minor changes to your .qmd. You had:

```
::: {.tabset-panel .column-page}
...
:::
```

I had to change this to

```
::: {.column-page}

::: {.tabset-panel}
...
:::

:::
```

(this change is necessary because of how we're handling tabsets now, and is not a bug on your said). With that said, after making those changes, I get this:

<img width="1725" alt="image" src="https://user-images.githubusercontent.com/285675/212491830-dabfafce-f229-41e4-9d7f-7d264d11b37c.png">

Can you provide the output of `quarto check`?

**Reply by howardm** [reply]
Thanks for testing and your continued interest!

1. The index.qmd for the page you're testing -- and the one I have just modified as you suggested -- is here: 

https://www.dropbox.com/s/0zjw93y6uo2yly3/index.qmd?dl=0

2. This image is what I see:

![TestResults](https://user-images.githubusercontent.com/6101226/212504690-87bfce30-8ca2-49b6-8b6b-8d089444dc7f.jpg)

___

3. Here is the information you requested:

```

^CHowardMannMacBook:QuartoThoracicImaging howardm$ quarto check

[✓] Checking Quarto installation......OK
      Version: 1.2.269
      Path: /Applications/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.7.4
      Path: /usr/local/opt/python/bin/python3.7
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

[✓] Checking R installation...........OK
      Version: 3.6.1
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Users/howardm/Library/R/3.6/library
        - /Library/Frameworks/R.framework/Versions/3.6/Resources/library
      rmarkdown: 2.18

[✓] Checking Knitr engine render......OK

```

Cheers.

**Reply by cscheid** [reply]
Can you try with the 1.3 prerelease version?

**Reply by howardm** [reply]
Sure.

I did  -- with exactly the same result as I show above.

Howard

**Reply by cscheid** [reply]
That's very strange. I can't reproduce it on my computer. (And I'm on macOS like you are.) Can you tell me the version reported in the output HTML file generated (specifically the `<meta name="generator" content="quarto-{VERSION}">` tag)?

**Reply by howardm** [reply]
Sure. Here's the top potion of that file:

```
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.107">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Thoracic Imaging</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>

```

Howard

**Reply by cscheid** [reply]
I'm totally stumped. I tried the solution I gave you on Firefox, Safari, and Chrome. None of them exhibit this issue.

Could you possibly be serving these from `file://` instead of `http://`? That's the only other thing I think could cause a problem.

**Reply by howardm** [reply]
@cscheid 

I've been viewing via: http://localhost:4444....

I'll deploy this to a Netlify server later this coming week. 

I'll let you know when I've done this. We'll see what it looks like. 

Thanks for all your interest and testing!

Howard

**Reply by cscheid** [reply]
Hm. that should work: it's what I'm doing.

I'm also seeing a dark background on your page that I don't see on mine. Could you have user stylesheets installed that are interfering here? What browser are you using?

**Reply by howardm** [reply]
@cscheid

Oh, I've just realized that this doesn't happen if I do not specify *any* form of Body Column extension:

![Screen Shot 2023-01-15 at 2 42 29 PM](https://user-images.githubusercontent.com/6101226/212569334-3c2725bf-2863-42a9-b8d5-273962e488b5.jpg)

Hah!  ??

But then, of course, there isn't room for the Tabset's content in the page.

Why does a succeeding Body Column extension produce this phenomenon ?

Cheers.

**Reply by howardm** [reply]
@cscheid

The dark background ? No, I just have light and dark themes. 

The issue is the same on both.

**Reply by dragonstyle** [reply]
Just FWIW this is working for me as well:

````markdown

<details><summary>Content to be revealed</summary>

Content that is revealed.

</details>


::: {.column-page}

::: {.panel-tabset}
## Clinical presentation

Pertinent clinical information is provided here.

## Radiography

![](images/CXR.jpg){.lightbox width="40%" fig-align="center"}

## CT

![](images/CT1.jpg){width="35%" fig-align="center"}

## CT Pacsbin

```{=html}
<div style="text-align:center;">

<iframe 
src="https://www.pacsbin.com/c/-ylmpOm7kd?s=6211650ea311df5697970ae9&i=6211650ea311df5697970afa&ww=1665&wc=-490&an=false&header=false&notes=false&overlay=false&scale=0.9" 
style="width: 475px; height: 475px; border: 0;"></iframe>

</div>
 
```
## Explanation

::: columns
::: {.column width="32%"}
![](images/CXR.jpg){.lightbox width="82%"}
:::

::: {.column width="32%"}
![](images/CT1.jpg){width="100%"}
:::

::: {.column width="2%"}
:::

::: {.column width="32%"}
Findings:

-   finding one [&xrarr;]{style="color: red"}
-   finding two
-   finding three
:::

:::

````

**Reply by howardm** [reply]
@dragonstyle 

Thanks.

It works for me, too. 

*But* does it work properly when the `Content that is revealed` consists of a multi-tab Tabset displayed using the  `.column-page` body extension functionality ?

That's the problem I'm encountering. 

I'm trying to understand your success        :-)

Howard

**Reply by dragonstyle** [reply]
Yeah that does not appear to work right now. I think is likely because the way our margin layout system works is by forwarding the grid system up through elements, and applying stylse to children to position the various elements as appropriate. In cases like collapsing elements (especially if you'd like the collapse UI to follow the body layout while the content actually follows some wider grid) we need to do special processing.

This is the simplest repro for me:

```markdown
---
title: "Test"
---


<details>
<summary>Content to be revealed</summary>

:::{.column-page}

hi this is page layout hi this is page layout hi this is page layout hi this is page layout hi this is page layout hi this is page layout hi this is page layout hi this is page layout hi this is page layout hi this is page layout hi this is page layout hi this is page layout

:::
</details>

```

**Reply by howardm** [reply]
Thanks for confirming and explaining that.

**Comment by howardm** [comment]
@cscheid 

Sure.  Thanks.
